string color
float64 app_x
float64 app_y
float64 app_z